### PR TITLE
ENG-9985: Ignore table row limit when applying DR data.

### PR DESCRIPTION
**Pull request by nshi**
- Ignore table row limit when applying DR data.
- Allow different table row limits/delete policies on master and replica clusters.

### CODE REVIEW

File: src/catgen/in/javasrc/DRCatalogDiffEngine.java [hunk 1]
```diff
@@ -34,7 +34,6 @@
  * 1. The localCatalog contains all DR tables contained in the remote catalog
  * 2. All shared DR tables contain the same columns in the same order
  * 3. All shared DR tables have the same partition column
- * 4. All shared DR tables have the same row limits/delete policies
  * 5. All shared DR tables have the same unique indexes/primary keys
```
Comment (luyangco) [inline src/catgen/in/javasrc/DRCatalogDiffEngine.java]: The comment needs to be reindexed to 4.

Comment (nshi) [inline src/catgen/in/javasrc/DRCatalogDiffEngine.java]: I just pushed the change.

File: src/catgen/in/javasrc/DRCatalogDiffEngine.java [hunk 2]
```diff
@@ -107,14 +113,9 @@ protected String checkModifyWhitelist(final CatalogType suspect, final CatalogTy
...
-            if ("cost".equals(field) ||
-                "seqscancount".equals(field) ||
-                "explainplan".equals(field) ||
-                "indexesused".equals(field) ||
-                "cachekeyprefix".equals(field)) {
-                return null;
-            }
-        } else if (isUniqueIndex(suspect) == isUniqueIndex(prevType) && !isUniqueIndexColumn(suspect)) {
+            return null;
+        } else if ((suspect instanceof Index && isUniqueIndex(suspect) == isUniqueIndex(prevType)) ||
```
Comment (luyangco) [inline src/catgen/in/javasrc/DRCatalogDiffEngine.java]: Should here to be `&&`?

Comment (nshi) [inline src/catgen/in/javasrc/DRCatalogDiffEngine.java]: No. The original conditions were very misleading, and it turned out to be wrong. I fixed it by splitting it into two conditions.

What it's trying to check are
* change of the uniqueness of an index, or
* change of columns inside an index

File: tests/ee/storage/DRBinaryLog_test.cpp [hunk 3]
```diff
@@ -1939,6 +1939,23 @@ TEST_F(DRBinaryLogTest, DeleteOverBufferLimit) {
     ASSERT_TRUE(false);
 }
 
+TEST_F(DRBinaryLogTest, IgnoreTableRowLimit) {
+    m_tableReplica->setTupleLimit(100);
+
+    const int total = 101;
```
Comment (luyangco) [inline tests/ee/storage/DRBinaryLog_test.cpp]: It's better to use a greater number to make the test more obvious, e.g. 200, so the reader doesn't have to calculate how many rows you insert into the table.

Comment (nshi) [inline tests/ee/storage/DRBinaryLog_test.cpp]: Will do.

**Comment by luyangco** [comment]
Review complete

**Pull request merged**